main.go

```diff
@@ -903,17 +966,58 @@ func transformLink(args []string) ([]string, error) {
 	return append(flags, paths...), nil
 }
 
+func splitFlagsFromArgs(all []string) (flags, args []string) {
```

Comment:
Can you explain again why we neeed this?

Reply:
Sure, I'll add some comments.